avocado/utils/memory.py

```diff
@@ -320,3 +320,56 @@ def get_thp_value(feature):
         return (re.search(r"\[(\w+)\]", value)).group(1)
     else:
         return value
+
+
+class MemInfo(object):
+    """
+    Representation of /proc/meminfo
+    """
+
+    class MemInfoItem(object):
```

Comment:
It's not wasted time if it solves a problem. This class has a data conversion aspect indeed, but I don't think we can call it a 'DataSize' object. We can implement one, which can actually be used by this `MemInfoItem` to make the byte unit conversions.
Apart from that, renaming this class to `DataSize` will confuse users, providing a class with a misleading name, considering the current implementation. What exactly are the problems you're trying to solve with your requests?